x/operator/keeper/avs_operator_shares.go

```diff
@@ -335,3 +335,35 @@ func (k *Keeper) GetStakerShare(ctx sdk.Context, avsAddr, stakerID, operatorAddr
 
 	return ret.Amount, nil
 }
+
+func (k *Keeper) GetAvgDelegatedValue(
+	ctx sdk.Context, operators []sdk.AccAddress, chainID, _ string,
+) ([]int64, error) {
+	avsAddr, err := k.avsKeeper.GetAvsAddrByChainID(ctx, chainID)
+	if err != nil {
+		return nil, err
+	}
+	ret := make([]int64, 0)
+	for _, operator := range operators {
+		share, err := k.GetOperatorShare(ctx, operator.String(), avsAddr)
+		if err != nil {
+			return nil, err
+		}
+		// truncate the USD value to int64
+		ret = append(ret, share.TruncateInt64())
+	}
+	return ret, nil
+}
+
+func (k *Keeper) GetOperatorAssetValue(ctx sdk.Context, operator sdk.AccAddress, chainID string) (int64, error) {
+	avsAddr, err := k.avsKeeper.GetAvsAddrByChainID(ctx, chainID)
```

Comment:
Is it necessary to validate the validity of the formatting rules for avsAddr and operatorAddr except for null?

- UpdateOperatorShare DeleteOperatorShare GetOperatorShare UpdateAVSShare
- UpdateStateForAsset DeleteAssetState (assetID, avsAddr)
- BatchUpdateShareForAVSAndOperator Does this function require validation of key-value pairs in the mapping avsOperatorChange?

Reply:
I will add code in the `optIn` function to check if the AVS address is an EVM contract address. Other checks are unnecessary because all entry points of calls will check the operator address type. If the type is incorrect, the relevant operator and AVS cannot be queried in the KV store. As for `avsOperatorChange`, value checks will be handled uniformly in `UpdateAssetDecValue`.